packages/react/src/IDE/IDE.tsx

```diff
@@ -129,17 +129,25 @@ type IDEChatProps = {
   'data-testid'?: string
 } & BaseProps<HTMLElement>
 
-type MessageRole = 'user' | 'assistant'
-
-export type IDEChatMessage = {
-  role: MessageRole
+type IDEChatMessageUser = {
+  role: 'user'
   handle: string
   avatar: string
   message: string
   codeSnippet?: string
-  highlighter?: 'hljs' // add additional highlighters as needed
+  highlighter?: 'hljs'
+}
+
+type IDEChatMessageAssistant = {
+  role: 'assistant'
+  handle: string
+  message: string
+  codeSnippet?: string
+  highlighter?: 'hljs'
```

Comment:
Nit... this seems a little duplicative. Took me a second to spot the difference . Worth creating an intersection with a shared message type for `handle`, `message`,  `codeSnippet` and `highlighter`?

Reply:
Works for me, updated 🙂